File: diag.py
import os
with open('diag_output.txt', 'w') as f:
    f.write(f"CWD: {os.getcwd()}\n")
    try:
        f.write(f"Contents of .: {os.listdir('.')}\n")
    except Exception as e:
        f.write(f"Error listing .: {e}\n")
    try:
        f.write(f"Contents of public: {os.listdir('public')}\n")
    except Exception as e:
        f.write(f"Error listing public: {e}\n")
    try:
        f.write(f"Contents of public/products: {os.listdir('public/products')}\n")
    except Exception as e:
        f.write(f"Error listing public/products: {e}\n")

File: diag.js
const fs = require('fs');
const path = require('path');

const log = (msg) => fs.appendFileSync('node_diag.txt', msg + '\n');

log(`CWD: ${process.cwd()}`);
try {
  log(`Contents of .: ${fs.readdirSync('.')}`);
} catch (e) {
  log(`Error listing .: ${e.message}`);
}
try {
  log(`Contents of public: ${fs.readdirSync('public')}`);
} catch (e) {
  log(`Error listing public: ${e.message}`);
}
try {
  log(`Contents of public/products: ${fs.readdirSync('public/products')}`);
} catch (e) {
  log(`Error listing public/products: ${e.message}`);
}
